sdk/core/azure-core/inc/azure/core/context.hpp

```diff
@@ -160,13 +275,14 @@ namespace Azure { namespace Core {
     }
 
     /**
-     * @brief Creates a context without a deadline, but with \p key and \p value associated with it.
+     * @brief Creates a new child context with \p key and \p value
+     * associated with it.
      *
      * @tparam T The type of the value to be stored with the key.
      * @param key A key to associate with this context.
      * @param value A value to associate with this context.
      *
-     * @return A child context with no deadline and the \p key and \p value associated with it.
```

Comment:
We should keep this clarification.
The child context, itself, has no deadline, which is useful to mention. Consider not removing this note.

If we want to clarify and mention something about the parent having a deadline and surpassing the deadline will cancel all children, in addition, that's reasonable.

Reply:
I removed a similar doccomment because it was flagged as misleading.

There is no way of distinguishing a context with a deadline and a context without a deadline as long as the context belongs in a chain of contexts. Because of this, describing context objects as having no deadline provides no value to the customer.

As a different variant on the same argument: the "WithDeadline" method doesn't say that the newly created context doesn't have a value associated with it, although a call to TryGetValue may succeed on that context.